WireDomain/Sources/WireDomain/Synchronization/IncrementalSync.swift

```diff
@@ -246,7 +246,9 @@ public struct IncrementalSync: IncrementalSyncProtocol {
             // If we need to abort, do it before processing the next batch.
             try Task.checkCancellation()
 
-            let envelopes = try await updateEventsStore.fetchStoredEventEnvelopes(limit: batchSize)
+            let envelopesWithObjectIDs = try await updateEventsStore.fetchStoredEventEnvelopes(limit: batchSize)
+            let envelopes = envelopesWithObjectIDs.map(\.0)
+            let envelopesObjectIDs = envelopesWithObjectIDs.map(\.objectID)
```

Comment:
thought: so there is another point here, if we fail to process the event we should not delete it